src/heaps/minmax_heap.jl

```diff
@@ -0,0 +1,261 @@
+        else
+            # bubble up min
+            _minmax_heap_bubble_up!(A, i, Forward)
+        end
+        
+    else
+        # max level
+        if i > 1 && A[i] < A[hparent(i)]
+            # swap to parent and bubble up min
+            tmp = A[i]
+            A[i] = A[hparent(i)]
+            A[hparent(i)] = tmp
+            _minmax_heap_bubble_up!(A, hparent(i), Forward)
+        else
+            # bubble up max
+            _minmax_heap_bubble_up!(A, i, Reverse)
+        end
+    end
+   return 
+end
+
+function _minmax_heap_bubble_up!(A::AbstractVector, i::Integer, o::Ordering, x=A[i])
+    gparent = hparent(hparent(i))
+    if i > hparent(i) > gparent >= 1
+        # i has grandparent
+        if lt(o, x, A[gparent])
+            A[i] = A[gparent]
+            A[gparent] = x
+            _minmax_heap_bubble_up!(A, gparent, o)
+        end
+    end
+    return          
+end
+
+function _minmax_heap_trickle_down!(A::AbstractVector, i::Integer)
+    if level(i) % 2 == 0
+        _minmax_heap_trickle_down!(A, i, Forward)
+    else
+        _minmax_heap_trickle_down!(A, i, Reverse)
+    end
+    return
+end
+
+function _minmax_heap_trickle_down!(A::AbstractVector, i::Integer, o::Ordering, x=A[i])
+    N = length(A)
+    if lchild(i) ≤ N
+        # tuple (min(val, j1) max(val, j2))
+        ext = extrema((A[j], j) for j in descendants(N, i))
+        m = lt(o, ext[1], ext[2]) ? ext[1][2] : ext[2][2]
+
+        if m ≥ 4*i
+            # this is a grandchild
+            if lt(o, A[m], A[i])
+                A[i] = A[m]
+                A[m] = x
+                if lt(o, A[hparent(m)], A[m])
+                    t = A[m]
+                    A[m] = A[hparent(m)]
+                    A[hparent(m)] = t
+                end
+                _minmax_heap_trickle_down!(A, m, o)
+            end
+        else
+            if lt(o, A[m], A[i])
+                A[i] = A[m]
+                A[m] = x
+            end
+        end
+    end
+    return
+end
+                    
+################################################
+#
+# utilities
+#
+################################################
+
+@inline level(i) = round(Int, floor(log2(i)))
+@inline lchild(i) = 2*i
+@inline rchild(i) = 2*i+1
+@inline children(i) = [lchild(i), rchild(i)]
+@inline hparent(i) = round(Int, floor(i/2))
+
+"""
+Return the indices of all children and grandchildren of
+position `i`.
+"""
+function descendants(N::T, i::T) where {T <: Integer}
+    _descendants = T[]
+    for child in children(i)
+        append!(_descendants, [child, lchild(child), rchild(child)])
+    end
+    return [d for d in _descendants if d ≤ N]
+end
+
+"""
+    is_minmax_heap(h::AbstractVector) -> Bool
+
+Return `true` if `A` is a min-max heap. A min-max heap is a
+heap where the minimum element is the root and the maximum
+element is a child of the root.
+"""
+function is_minmax_heap(A::AbstractVector)
+
+    isheap = true
+    N = length(A)
+
+    for i = 1:N
```

Comment:
For loops should be written with
`for i in 1:N`
the use of `=` instread of `in` is an ugly matlabism.

Changing this makes it consistant with the other loops in the package and this PR.

Reply:
Resolved in [00b29b4](https://github.com/JuliaCollections/DataStructures.jl/pull/468/commits/00b29b44b13cc7bb3f3fcf7c49260a74de9fadd6)